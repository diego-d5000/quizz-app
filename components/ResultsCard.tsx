import truncate from 'lodash/truncate';
import he from 'he';
import cn from 'classnames';
import styles from './ResultsCard.module.scss';

const truncateQuestion = (text) => truncate(text, { length: 70, separator: ' ' });

const ResultsCard = ({ questions, answers }: ResultsCardProps) => {
  return (
    <article className="bg-white rounded-lg p-4 pl-8 text-black max-h-full">
      <ol className="space-y-2 list-decimal">
        {questions.map((q, i) => {
          return (
            <li
              className={cn(
                'leading-tight break-words',
                { [styles['wrong-answer']]: q.correct_answer !== answers[i] },
                { [styles['correct-answer']]: q.correct_answer === answers[i] }
              )}
            >
              {truncateQuestion(he.decode(q.question))}
            </li>
          );
        })}
      </ol>
    </article>
  );
};

interface ResultsCardProps {
  questions?: any;
  answers?: any;
}

export default ResultsCard;
