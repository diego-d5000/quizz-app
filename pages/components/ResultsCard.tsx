import truncate from 'lodash/truncate';
import unescape from 'lodash/unescape';
import cn from 'classnames';
import styles from './ResultsCard.module.scss';

const truncateQuestion = (text) => truncate(text, { length: 70, separator: ' ' });

const ResultsCard = ({ questions, answers }: ResultsCardProps) => {
  return (
    <article className="bg-white rounded-lg p-4 pl-8 text-black max-h-full">
      <ol className="space-y-2 list-decimal">
        {questions.map((q, i) => {
          return (
            <li
              className={cn(
                'leading-tight',
                { [styles['wrong-answer']]: q.correct_answer !== answers[i] },
                { [styles['correct-answer']]: q.correct_answer === answers[i] }
              )}
            >
              {truncateQuestion(unescape(q.question))}
            </li>
          );
        })}
      </ol>
    </article>
  );
};

interface ResultsCardProps {
  questions?: any;
  answers?: any;
}

export default ResultsCard;
