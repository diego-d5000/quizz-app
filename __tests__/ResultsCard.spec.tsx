import React from 'react';
import { render, act, waitFor } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ResultsCard from '../components/ResultsCard';
import mockData from '../mockdata';

describe('ResultsCard Component', () => {
  const { results: mockQuestions } = mockData;
  const mockAnswers = mockQuestions.map((q, i) =>
    i % 2 == 0 ? q.correct_answer : q.incorrect_answers[0]
  );

  it('snapshot', () => {
    const tree = renderer
      .create(<ResultsCard questions={mockQuestions} answers={mockAnswers} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
