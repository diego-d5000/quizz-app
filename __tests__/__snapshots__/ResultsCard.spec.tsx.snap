// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsCard Component snapshot 1`] = `
<article
  className="bg-white rounded-lg p-4 pl-8 text-black max-h-full"
>
  <ol
    className="space-y-2 list-decimal"
  >
    <li
      className="leading-tight break-words correct-answer"
    >
      Unturned originally started as a Roblox game.
    </li>
    <li
      className="leading-tight break-words wrong-answer"
    >
      The binary number "101001101" is equivalent to the Decimal number...
    </li>
    <li
      className="leading-tight break-words correct-answer"
    >
      In "Portal 2", Cave Johnson started out Aperture Science as a...
    </li>
    <li
      className="leading-tight break-words wrong-answer"
    >
      The fourth funnel of the RMS Titanic was fake designed to make the...
    </li>
    <li
      className="leading-tight break-words correct-answer"
    >
      "Cube", "Cube 2: Hypercube" and "Cube Zero" were directed by the...
    </li>
    <li
      className="leading-tight break-words wrong-answer"
    >
      Harry Potter was born on July 31st, 1980.
    </li>
    <li
      className="leading-tight break-words correct-answer"
    >
      The weapon Clint Eastwood uses in "Dirty Harry" was a .44 Automag.
    </li>
    <li
      className="leading-tight break-words wrong-answer"
    >
      Spoon theory is a theory, utilizing "Spoons" as a metaphor for...
    </li>
    <li
      className="leading-tight break-words correct-answer"
    >
      In the "Kagerou Daze" series, Shintaro Kisaragi is prominently...
    </li>
    <li
      className="leading-tight break-words wrong-answer"
    >
      The singer Billie Holiday was also known as "Lady Day".
    </li>
  </ol>
</article>
`;
